diff -rupN a/bin/make-base-vm b/bin/make-base-vm
--- a/bin/make-base-vm
+++ b/bin/make-base-vm
@@ -72,8 +72,8 @@ if [ $DISTRO = "debian" -a $LXC = "1" ];
 fi
 
 if [ $DISTRO = "ubuntu" ]; then
-  MIRROR=$MIRROR_BASE/archive.ubuntu.com/ubuntu
-  SECURITY_MIRROR=$MIRROR_BASE/security.ubuntu.com/ubuntu
+  MIRROR=$MIRROR_BASE/old-releases.ubuntu.com/ubuntu
+  SECURITY_MIRROR=$MIRROR_BASE/old-releases.ubuntu.com/ubuntu
 elif [ $DISTRO = "debian" ]; then
   MIRROR=$MIRROR_BASE/ftp.debian.org/debian
   SECURITY_MIRROR=$MIRROR_BASE/security.debian.org/
@@ -106,6 +106,7 @@ if [ $DISTRO = "debian" ]; then
 fi
 
 addpkg=pciutils,build-essential,git-core,subversion,$LOCALE_PKG,wget,lsb-release
+removepkg=cron
 
 if [ $DISTRO = "ubuntu" ]; then
   # Need comma at end to work around an issue with apt for Debian <= Wheezy regarding empty strings
diff -rupN a/libexec/config-bootstrap-fixup b/libexec/config-bootstrap-fixup
--- a/libexec/config-bootstrap-fixup
+++ b/libexec/config-bootstrap-fixup
@@ -4,7 +4,7 @@ wd=`pwd`
 
 # Default host IP is 10.0.2.2
 if [ -z "$GITIAN_HOST_IP" ]; then
-  GITIAN_HOST_IP=10.0.2.2
+  GITIAN_HOST_IP=10.0.2.15
 fi
 
 # Default to host IP if the MIRROR_HOST is not set, or set to 127.0.0.1
diff -rupN a/target-bin/bootstrap-fixup.in b/target-bin/bootstrap-fixup.in
--- a/target-bin/bootstrap-fixup.in
+++ b/target-bin/bootstrap-fixup.in
@@ -6,9 +6,9 @@ DISTRIB_NAME=`lsb_release -is`
 DISTRIB_CODENAME=`lsb_release -cs`
 
 if [ $DISTRIB_NAME = "Ubuntu" ]; then
-  echo "deb http://HOSTIP:3142/archive.ubuntu.com/ubuntu $DISTRIB_CODENAME main universe" > $1/etc/apt/sources.list
-  echo "deb http://HOSTIP:3142/security.ubuntu.com/ubuntu $DISTRIB_CODENAME-security main universe" >> $1/etc/apt/sources.list
-  echo "deb http://HOSTIP:3142/archive.ubuntu.com/ubuntu $DISTRIB_CODENAME-updates main universe" >> $1/etc/apt/sources.list
+  echo "deb http://HOSTIP:3142/old-releases.ubuntu.com/ubuntu $DISTRIB_CODENAME main universe" > $1/etc/apt/sources.list
+  echo "deb http://HOSTIP:3142/old-releases.ubuntu.com/ubuntu $DISTRIB_CODENAME-security main universe" >> $1/etc/apt/sources.list
+  echo "deb http://HOSTIP:3142/old-releases.ubuntu.com/ubuntu $DISTRIB_CODENAME-updates main universe" >> $1/etc/apt/sources.list
 elif [ $DISTRIB_NAME = "Debian" ]; then
   echo "deb http://HOSTIP:3142/ftp.debian.org/debian $DISTRIB_CODENAME main" > $1/etc/apt/sources.list
   echo "deb http://HOSTIP:3142/security.debian.org/ $DISTRIB_CODENAME/updates main" >> $1/etc/apt/sources.list
diff -rupN a/target-bin/upgrade-system.sh b/target-bin/upgrade-system.sh
--- a/target-bin/upgrade-system.sh
+++ b/target-bin/upgrade-system.sh
@@ -2,7 +2,7 @@
 
 # Upgrade system
 
-set -e
+#set -e
 
 mkdir -p /var/cache/gitian
 
